wasm/module_test.go

```diff
@@ -6,6 +6,45 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+func TestDecodeModule(t *testing.T) {
+	tests := []struct {
+		name        string
+		input       []byte
+		expected    *Module
+		expectedErr string
+	}{
+		{
+			name:     "empty",
+			input:    []byte("\x00asm\x01\x00\x00\x00"),
+			expected: &Module{CustomSections: map[string][]byte{}},
+		},
+		{
+			name:        "wrong magic",
+			input:       []byte("wasm\x01\x00\x00\x00"),
+			expectedErr: "invalid magic number",
+		},
+		{
+			name:        "wrong version",
+			input:       []byte("\x00asm\x01\x00\x00\x01"),
+			expectedErr: "invalid version header",
+		},
+	}
+
+	for _, tc := range tests {
+		tc := tc // pin! see https://github.com/kyoh86/scopelint for why
```

Comment:
I think we don't necessarily add this comment everywhere :D
```suggestion
		tc := tc
```

Reply:
ok if no comment, I'll switch slightly to ensure compile fail.